crates/stackable-webhook/src/webhooks/conversion_webhook.rs

```diff
@@ -0,0 +1,201 @@
+                        path: Some(format!("/convert/{crd_name}")),
+                        port: Some(options.socket_addr.port().into()),
+                    }),
+                    // Here, ByteString takes care of encoding the provided content as base64.
+                    ca_bundle: Some(new_ca_bundle.to_owned()),
+                    url: None,
+                }),
+            }),
+        });
+
+        // Deploy the updated CRDs using a server-side apply.
+        let patch = Patch::Apply(&crd);
+
+        // We force apply here, because we want to become the sole manager of the CRD. This
+        // avoids any conflicts from previous deployments via helm or stackablectl which are
+        // reported with the following error message:
+        //
+        // Apply failed with 2 conflicts: conflicts with "stackablectl" using apiextensions.k8s.io/v1:
+        //   - .spec.versions
+        //   - .spec.conversion.strategy: Conflict
+        //
+        // The official Kubernetes documentation provides three options on how to solve
+        // these conflicts. Option 1 is used, which is described as follows:
+        //
+        // Overwrite value, become sole manager: If overwriting the value was intentional
+        // (or if the applier is an automated process like a controller) the applier should
+        // set the force query parameter to true [...], and make the request again. This
+        // forces the operation to succeed, changes the value of the field, and removes the
+        // field from all other managers' entries in managedFields.
+        //
+        // See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
+        let patch_params = PatchParams::apply(&self.field_manager).force();
+
+        crd_api
+            .patch(&crd_name, &patch_params, &patch)
+            .await
+            .with_context(|_| PatchCrdSnafu { crd_name })?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl<H> Webhook for ConversionWebhook<H>
+where
+    H: FnOnce(ConversionReview) -> ConversionReview + Clone + Send + Sync + 'static,
+{
+    fn register_routes(&self, mut router: Router) -> Router {
+        for (crd, handler) in self.crds_and_handlers.clone() {
+            let crd_name = crd.name_any();
+            let handler_fn = |Json(review): Json<ConversionReview>| async {
+                let review = handler(review);
+                Json(review)
+            };
+
+            let route = format!("/convert/{crd_name}");
+            router = router.route(&route, post(handler_fn));
+        }
+
+        router
+    }
+
+    #[instrument(skip(self))]
+    async fn handle_certificate_rotation(
+        &mut self,
+        _new_certificate: &Certificate,
+        new_ca_bundle: &ByteString,
+        options: &WebhookServerOptions,
+    ) -> Result<(), WebhookError> {
+        if self.disable_crd_maintenance {
+            return Ok(());
+        }
```

Comment:
suggestion: We should maybe add a `is_enabled` function to the `Webhook` trait to basically skip adding this particular webhook when handling the rotation at the top-level `Webhookserver`.

Reply:
We should not call it `is_eanbled`, as the webhook itself is always enabled. The question is if it does something on cert rotation.
Added in 85481e8, although not really sure if this improve readability

Reply:
Yeah, this doesn't improve a whole lot. I'll let you decide which approach you like better.